plugin/src/test/java/io/jenkins/plugins/casc/core/JenkinsConfiguratorCloudSupportTest.java

```diff
@@ -0,0 +1,145 @@
+package io.jenkins.plugins.casc.core;
+
+import hudson.Extension;
+import hudson.model.Descriptor;
+import hudson.model.Node;
+import hudson.model.TaskListener;
+import hudson.slaves.AbstractCloudComputer;
+import hudson.slaves.AbstractCloudSlave;
+import hudson.slaves.EphemeralNode;
+import io.jenkins.plugins.casc.ConfigurationAsCode;
+import io.jenkins.plugins.casc.misc.ConfiguredWithCode;
+import io.jenkins.plugins.casc.misc.JenkinsConfiguredWithCodeRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.jvnet.hudson.test.PretendSlave;
+
+import java.io.IOException;
+import java.util.ArrayList;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+public class JenkinsConfiguratorCloudSupportTest {
+
+    @Rule
+    public JenkinsConfiguredWithCodeRule j = new JenkinsConfiguredWithCodeRule();
+
+    @Test
+    @ConfiguredWithCode("JenkinsConfiguratorCloudSupportTest.yml")
+    public void should_have_nodes_configured() {
+        assertEquals("Base nodes not found", 2, j.jenkins.getNodes().size());
+    }
+
+    @Test
+    public void should_remove_normal_nodes_configured_after_reload() throws Exception {
+        final Node slave = new StaticPretendSlave();
+        j.jenkins.addNode(slave);
+
+        ConfigurationAsCode.get().configure(this.getClass().getResource("JenkinsConfiguratorCloudSupportTest.yml").toString());
+        assertEquals("Base nodes not found", 2, j.jenkins.getNodes().size());
+    }
+
+    @Test
+    public void should_keep_cloud_no_instantiable_nodes_configured_after_reload() throws Exception {
```

Comment:
could you add a test for exporting the config to make sure it exports what is expected please?
Something like:
https://github.com/jenkinsci/configuration-as-code-plugin/pull/784/files#diff-63c3227aa81b1bacf4cec3ad0f7274a6R74

Reply:
Added export test..
Was there any reason not to use spock or some similar tool in tests..
JUnit after using spock for a while is really `bulky` :)

Reply:
I know some plugins use Spock and if it makes sense, I'd welcome it. Can we reuse the test classes that are in `misc` package?

Perhaps best done in a separate PR.

Reply:
I think that migration right now will not be that strait forward.. 
`spock` => `groovy` :)

Things from `misc` would work as a charm...

Reply:
And for sure it needs extra PR, if ever :)

Reply:
I don't mind groovy, we live and breath it everyday in Jenkins pipelines :sweat_smile: